### ISSUE TITLE
Client Result Cache

### ISSUE THREAD

**Issue by sandrozbinden** [oracle/node-oracledb#1368]
### Problem ### 
Hi there 

I try to get the oracle client result cache working without any modifications on the oracle database server. 
According to the documentation [(Client Result Cache)](http://oracle.github.io/node-oracledb/doc/api.html#clientresultcache) the cache can be enabled by creating a `oraaccess.xml` configuration file and adding a `/*+ result_cache */` hint on the sql query

However I does not seem to have an impact on the query execution time. 
So my questions:
-  Did I make a mistake initializing the `configDir` within the `initOracleClient` method?
-  Does Client Result Cache work for queries without a preparedStatement or preparedCall?
-  What is the best way to check if the query has actually been cached in client memory?
-  Does anybody know where I could find a working example?

Any help much apprechiated

### oraaccess.xml ### 
```xml 
<?xml version="1.0"?>
<oraaccess xmlns="http://xmlns.oracle.com/oci/oraaccess"
           xmlns:oci="http://xmlns.oracle.com/oci/oraaccess"
           schemaLocation="http://xmlns.oracle.com/oci/oraaccess
  http://xmlns.oracle.com/oci/oraaccess.xsd">
    <default_parameters>
        <prefetch>
            <rows>50</rows>
        </prefetch>
        <statement_cache>
            <size>100</size>
        </statement_cache>
        <result_cache>
            <max_rset_rows>100</max_rset_rows>
            <max_rset_size>10K</max_rset_size>
            <max_size>64M</max_size>
        </result_cache>
    </default_parameters>
</oraaccess>
```

### NodeJS Code ### 

```typescript
  import oracledb, { BindParameters, Result } from 'oracledb';
  import { logger } from '../utils/logger';
  
  oracledb.initOracleClient({configDir: 'C:\\dev\\database\\config',}); //oraaccess.xml is inside config directory
  oracledb.fetchAsString = [oracledb.CLOB, oracledb.DATE];
  oracledb.fetchArraySize = 5000;
  oracledb.autoCommit = true;
  
  export async function createConnectionPool(): Promise<boolean> {
      const connectionSettings = createConnectionSettings();
      try {
          await oracledb.createPool(connectionSettings);
          logger.info('Connection pool started');
          return true;
      } catch (err) {
          logger.error('createConnectionPool() error: ' + err.message);
          return false;
      }
  }

export async function executeQuery<T = unknown>(sql: string, bindParams: BindParameters): Promise<Result<T>> {
    logger.debug(`ExecuteQuery ${sql} with bindParams: ${JSON.stringify(bindParams)}`);
    const conn = await oracledb.getPool().getConnection();
    return conn
        .execute<T>(sql, bindParams, { outFormat: oracledb.OUT_FORMAT_OBJECT })
        .catch((e) => {
            logger.error(
                `Error executeQuery ${sql} with bindParams: ${JSON.stringify(bindParams)} with exeception ${e}`
            );
            throw e;
        })
        .finally(() => conn.close());
}

export async function loadPatient(id: string): Promise<Patient> {
    const sql = 'SELECT /*+ result_cache */ * FROM PATIENT where id= :id';
    const result: Result<PatientModel> = await executeQuery(sql, { id });
    const dbPatient = result.rows ? result.rows[0] : undefined;
    if (!dbPatient) {
        throw new Error(`Patient with id ${id} does not exist`);
    }
    return mapPatientModelToPatient(dbPatient);
}
```

### Version ### 

| Description            | Version                                                                      |
|------------------------|------------------------------------------------------------------------------|
| oracledb.versionString | 5.1.0                                                                        |
| oracle                 | Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production |
| process.platform       | win32                                                                        |
| process.version       |  v12.18.3                                                                        |
| process.arch        | x64                                                                        |
| oracledb.oracleClientVersionString        |  19.8.0.0.0                                                                        |

**Comment by cjbj** [comment]
The [oraacess.xml doc](https://docs.oracle.com/en/database/oracle/oracle-database/21/lnoci/build-and-configure-oci-applications.html#GUID-4B68F0BB-16FA-40C6-BA13-CFD9A2B110A8) says that its settings override any sqlnet.ora settings.  SQL*Net doc seems mostly AWOL, but [here](https://docs.oracle.com/en/database/oracle/oracle-database/19/adfns/performance-and-scalability.html#GUID-5BF0E0FC-BF34-4B73-8832-86268886A092) it says:

> If client result cache is enabled on the server by CLIENT_RESULT_CACHE_SIZE, then its value can be overridden by the sqlnet.ora configuration parameter OCI_RESULT_CACHE_MAX_SIZE. If client result cache is disabled on the server, then OCI_RESULT_CACHE_MAX_SIZE is ignored and client result cache cannot be enabled on the client.

In summary, I believe you will need to enable CRC on the DB server.

I'll update node-oracledb's use of '[alternatively](http://oracle.github.io/node-oracledb/doc/api.html#clientresultcache)', and also see about some improvements in the SQL\*Net doc - it's possible these didn't end up being SQL*Net parameters and the doc references that say 'SQL\*Net' should say 'OCI'.

Regarding verifying CRC, the traditional way is to check the number of statement executions the DB performed.  If CRC is enabled, then the DB never gets sent the statement and doesn't execute it.

In SQL*Plus I did this:
```
ALTER SYSTEM SET CLIENT_RESULT_CACHE_LAG = 3000 SCOPE=SPFILE;
ALTER SYSTEM SET CLIENT_RESULT_CACHE_SIZE = 64K SCOPE=SPFILE;
STARTUP FORCE
```
And then ran [crc.js](https://gist.github.com/cjbj/b147c04cf9b85a2384e5a1a17ba00064) which runs a pair of statements 100 times:
```
    const q = `select id from crc_tab`;
    const qc = `select /*+ result_cache */ id from crc_tab`;
```

The output was:
```
No CRC 100 executions
CRC 1 executions
```
If I tried your oraacess.xml files without the ALTER SYSTEM commands, I got a result showing that CRC wasn't enabled:
```
No CRC 100 executions
CRC 100 executions
```
